### PR TITLE
Add localization

### PR DESCRIPTION
**Pull request by misha130**
### Pull Request summary

- Adds Translations
- Adds a basic culture ("en") with a default

### Link to Project Card/Issue

https://github.com/codidact/core/issues/63

### Pull Request Tasklist

- [x] Pull request references an issue
- [x] Branch follows contributing guidelines format
- [x] Pull request targets the `develop` branch

### CODE REVIEW

**Review by asynts — COMMENTED** [review]
This looks good, but I disagree with the logic that chooses which culture is used to represent the page.

*(The following is my opinion.)*

My understanding was that the canonical communities are in English. Therefor, there should not possible to change the interface language to something else because this would encourage people to create posts in that language.

However, there would be communities where this would make sense. Maybe a Spanish community or something of the like. 

The logic that chooses the culture is determined by the [`RequestLocalizationOptions.RequestCultureProviders`][1] property. In my opinion the defaults are not what we want, there should be two implementations of `IRequestCultureProvider`:

  - `CommunityConstantCultureProvider` this would set the culture to a constant which we would configure with the `appsettings.json` file. This provider would be used for the English communities. 

  - `CommunityChoiceCultureProvider` this would allow a user to choose a language from a list of allowed languages. For a Spanish community we would allow the choices "English" and "Spanish" for example.

## Addendum

Actually, both could be done in a single implementation. Just read an array from the configuration file and if there is only one option, make it constant.

  [1]: https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.builder.requestlocalizationoptions.requestcultureproviders?view=aspnetcore-3.1#Microsoft_AspNetCore_Builder_RequestLocalizationOptions_RequestCultureProviders

**Comment by misha130** [comment]
> Actually, both could be done in a single implementation. Just read an array from the configuration file and if there is only one option, make it constant.

So in general if I switch the configuration to read app.settings for the supported cultures it would be okay? Without implementing a `IRequestCultureProvider`? Or do I need the impl to stop people from switching on the constant

**Comment by asynts** [comment]
> So in general if I switch the configuration to read app.settings for the supported cultures it would be okay? Without implementing a IRequestCultureProvider? Or do I need the impl to stop people from switching on the constant

That's even simpler, that should work.

File: src/WebApp/Startup.cs [hunk 1]
```diff
@@ -97,6 +123,9 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env, ILogger<
 
             app.UseRouting();
 
+            var localizationOptions = app.ApplicationServices.GetService<IOptions<RequestLocalizationOptions>>().Value;
+            app.UseRequestLocalization(localizationOptions);
```
Comment (asynts) [inline src/WebApp/Startup.cs]: I think you can just use the following, it should pick up the options automatically. (I can't test this right now.)

~~~csharp
app.UseRequestLocalization();
~~~

File: src/WebApp/Startup.cs [hunk 2]
```diff
@@ -73,7 +77,29 @@ public void ConfigureServices(IServiceCollection services)
...
                 .AddRazorRuntimeCompilation()
-                .AddFluentValidation();
+                .AddFluentValidation()
+                .AddViewLocalization(options => options.ResourcesPath = "Resources");
+
+            services.Configure<RequestLocalizationOptions>(
+             opts =>
+            {
+                var settingSupportedCultures = Configuration.GetSection("Localization:Cultures")
+                                                .GetChildren().Select(child => child.Value);
```
Comment (asynts) [inline src/WebApp/Startup.cs]: I can't test this right now, but it should be possible to do this as follows:

~~~csharp
var settingSupportedCultures = Configuration.GetValue<string[]>("Localization:Cultures");
~~~

Comment (misha130) [reply src/WebApp/Startup.cs]: Nope. It was the first thing I tried too.
Making a class is an overkill for now

File: src/WebApp/Startup.cs (hunk 1 above)
Comment (asynts) [reply src/WebApp/Startup.cs]: I've tested this, it is able to pick up the options from the services collection.
```suggestion
            app.UseRequestLocalization();
```

File: src/WebApp/Startup.cs (hunk 2 above)
Comment (asynts) [reply src/WebApp/Startup.cs]: I've tried this now and you are correct, but the following works:

```suggestion
                var settingSupportedCultures = Configuration.GetSection("Localization:Cultures").Get<string[]>();
```

**Review by asynts — APPROVED** [review]
I've tested this on my machine and it works. (There was an issue but it was unrelated to this pull request, I created #90 for this.)

File: src/WebApp/Startup.cs (hunk 2 above)
Comment (misha130) [reply src/WebApp/Startup.cs]: Very cool. Couldn't figure it out myself

**Review by luap42 — APPROVED** [review]

**Pull request merged**